### DISCUSSION TITLE
Allow possibility to provide typings for custom elements while consuming them in Vue Template

### DISCUSSION THREAD

**Discussion by chintankavathia** [vuejs/rfcs#673] [General]
Currently when we use custom elements in Vue Template which are built outside of Vue. We don't get any typing support for these custom elements. In React we can do this by just extending `JSX IntrinsicElements` interface. 

```ts
namespace JSX {
    interface IntrinsicElements {
      "custom-element": JSX.HTMLAttributes<CustomElement>;
    }
  }
```

I would expect that Vue should also provide such a way so that custom elements types can be provided and Vue Template can recognise them.

**Comment by zhiyuanzmj** [comment]
```ts
declare module '@vue/runtime-dom' {
  interface HTMLAttributes {
    [key: string]: any
    'v-permission'?: string
    'v-loading'?: boolean
  }
}

declare module '@vue/runtime-core' {
  interface AllowedComponentProps {
    'v-permission'?: string
    'v-loading'?: boolean
  }
}
```

**Reply by chintankavathia** [reply]
I guess this will add `v-loading`, `v-permission` as possible property to all components. which doesn't help as I need to define diff props to diff web components individually.
